addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -296,12 +302,33 @@ def _get_carryover_date(self, date_from):
             carryover_date += relativedelta(years=1)
         return carryover_date
 
-    def _add_days_to_allocation(self, current_level, current_level_maximum_leave, leaves_taken, period_start, period_end):
+    def _add_days_to_allocation(self, allocation_data):
```

Comment:
Hello hello
making this will make the method quite unclear IMO so I would at least add a docstring to explicit what allocation_data really is to not enforce people to read the totality of the method to understand what they should put in the data and or use *args or **kwargs that you can assign to all the vars at the the first line of the method because at some point maybe someone will want to use this method

or if you don't want people to use it you need to make it a closure of _process_accrual_plans